intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,754 @@
+    "    )\n",
+    "\n",
+    "    del persisted_train_x_y\n",
+    "    gc.collect()\n",
+    "\n",
+    "    dmat_time = time.time() - t1\n",
+    "    print(\"Prepared data for XGB training\")\n",
+    "\n",
+    "    print(\"Training model\")\n",
+    "    t1 = time.time()\n",
+    "\n",
+    "    print(\"XGB training for part_count:{}\".format(part_count))\n",
+    "    bst = xgb.dask.train(\n",
+    "        client, dxgb_gpu_params, dmat, num_boost_round=NUM_BOOST_ROUND,\n",
+    "    )\n",
+    "\n",
+    "    train_time = time.time() - t1\n",
+    "    print(\"Training complete\")\n",
+    "    return (bst, dmat_time, train_time)\n",
+    "\n",
+    "\n",
+    "def run_etl(start_year, end_year, data_dir, client):\n",
+    "\n",
+    "    print(\"Starting ETL\")\n",
+    "    t1 = time.time()\n",
+    "\n",
+    "    perf_data_path = data_dir + \"perf/\"\n",
+    "\n",
+    "    gpu_dfs = []\n",
+    "    quarter = 1\n",
+    "    year = start_year\n",
+    "    count = 0\n",
+    "    while year <= end_year:\n",
+    "        for file in glob(\n",
+    "            os.path.join(\n",
+    "                perf_data_path + \"/Performance_\" + str(year) + \"Q\" + str(quarter) + \"*\"\n",
+    "            )\n",
+    "        ):\n",
+    "            gpu_dfs.append(\n",
+    "                process_quarter_gpu(\n",
+    "                    year=year, quarter=quarter, perf_file=file, client=client\n",
+    "                )\n",
+    "            )\n",
+    "            count += 1\n",
+    "        quarter += 1\n",
+    "        if quarter == 5:\n",
+    "            year += 1\n",
+    "            quarter = 1\n",
+    "    print(\"ETL for start_year:{} and end_year:{}\".format(start_year, end_year))\n",
+    "    wait(gpu_dfs)\n",
+    "\n",
+    "    etl_time = time.time() - t1\n",
+    "\n",
+    "    print(\"ETL done!\")\n",
+    "    return (gpu_dfs, etl_time)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if __name__ == \"__main__\":\n",
+    "\n",
+    "    import cudf\n",
+    "    import xgboost as xgb\n",
+    "    import dask_xgboost as dxgb_gpu\n",
+    "    import dask_cudf\n",
+    "\n",
+    "    cmd = \"hostname --all-ip-addresses\"\n",
+    "    process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)\n",
+    "    output, error = process.communicate()\n",
+    "    IPADDR = str(output.decode()).split()[0]\n",
+    "\n",
+    "    cluster = LocalCUDACluster(ip=IPADDR)\n",
+    "    client = Client(cluster)\n",
+    "\n",
+    "    data_dir = os.environ.get(\"MORTGAGE_DATA_DIR\", \"\")\n",
+    "    res = client.run(memory_info)\n",
+    "    total_mem = sum(res.values())\n",
+    "    min_mem = min(res.values())\n",
+    "\n",
+    "    start_year, end_year, part_count, use_1GB_splits = determine_dataset(\n",
+    "        total_mem=total_mem, min_mem=min_mem, part_count=os.environ.get(\"part_count\")\n",
```

Comment:
This looks like it assumes the user set this env variable. What should we default to if they didn?

Same request for clarifying comments here about what part count is, how it's used relative to ETL number of quarters.

Reply:
If the user doesn't set this, it defaults to `None` in which case we rely on the auto selection, instead of explicitly forcing a specific part_count (and by construction `start_year`, `end_year`)

Reply:
Added comments to explain this a bit